libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -147,6 +159,22 @@ internal class RouteRefreshController(
         }
     }
 
+    private suspend fun refreshRoutesOrNull(
+        routes: List<NavigationRoute>
+    ): List<NavigationRoute?> {
+        return coroutineScope {
+            routes.map { route ->
+                async {
+                    withTimeoutOrNull(routeRefreshOptions.intervalMillis) {
+                        refreshRouteOrNull(
```

Comment:
Do I understand correctly that for each route a separate network request is executed? Isn't there too much overhead? It seems like it would be more optimal to be able to do one request for several routes.

Reply:
I agree. Refreshing a few routes in one request would also help us avoid dealing with the situation from the [discussion](https://github.com/mapbox/mapbox-navigation-android/pull/5923#discussion_r895923903). But Directions API won't support it in the nearest future 😞 Anyway, we will be able to get rid of parallel requests as soon as they implement it later 🙂